### ISSUE TITLE
Found incorrect code within "Selectively disable Related Posts from appearing on specific posts" on Jetpack Support

### ISSUE THREAD

**Issue by interglobalmedia** [Automattic/jetpack#2489]
Hi,

I was adding code to one of my websites today that I took from http://jetpack.me/support/related-posts/customize-related-posts/#disable, and it didn't work. When I looked more closely at the code, I realized that there was a mistake. I fixed it, and it works beautifully. This is what appears on http://jetpack.me/support/related-posts/customize-related-posts/#disable:

```php
function jetpackme_no_related_posts( $options ) {
    if ( is_single( array( 17, 19, 1, 11 ) ) ) {
        $options['enabled'] = false;
    }
    return $options;
}
add_filter( 'jetpack_relatedposts_filter_options', 'jetpackme_no_related_posts' );
```

 This is what the code for "disabling Related Posts" from appearing on specific posts should be:

```php
function jetpackme_no_relatedposts( $options ) {
    if ( is_single( array(378, 397, 434 ) ) ) {
        $options['enabled'] = false;
    }
    return $options;
}
add_filter( 'jetpack_relatedposts_filter_options', 'jetpackme_no_relatedposts');
```

I am new to this, and wasn't quite sure how to go about contributing this fix, because it is not actually part of the plugin itself. If there is a better way to do this, please let me know! I just know that a lot of people in the support forums were having issues with customizing Jetpack, and specifically related posts, so I thought it important to share. Thanks!

**Comment by jeherve** [comment]
I'm afraid I don't see a difference between the 2 versions of the snippet. Could you tell me what the differences are?

Thanks!

**Comment by interglobalmedia** [comment]
Hi Jeremy,

I also had that problem initially. I couldn't figure out why the code isn't working, but there is a difference, be it ever so slight, that makes the difference from working and not working. Do you see how on line 1 of the original code, the function line, and the final line, the add_filter line, related_posts has an underscore between related and posts. In my version, in the function line and the add_filter line, it is simply relatedposts, no underscore. That's the difference that makes the difference between working and not working. Hope this helps!

**Comment by jeherve** [comment]
> Do you see how on line 1 of the original code, the function line, and the final line, the add_filter line, related_posts has an underscore between related and posts.

The name of the function does not matter, as long as you use the same function name when adding it to the filter. It could be `jetpackme_no_relatedposts`, `jetpackme_no_relatedposts`, or `interglobalmedia_thing`.

Is it possible that you used a different function name when adding the function to the `jetpack_relatedposts_filter_options` filter?

I just tested the code again, and it appears to work properly.

**Comment by interglobalmedia** [comment]
No I did not.

**Comment by jeherve** [comment]
Could you give it another try, using the exact code you pasted above? That's the code I used in my test.

**Comment by interglobalmedia** [comment]
Hi Jeremy,

You are right that the add_filter as you had it should have been able to work, but for whatever reason did not on my staging or live production site. And I had copied the code right into my functions.php file. I talked to another WordPress developer about it tonight. It remains a mystery! However, we did come up with the solution I was really looking for in the first place, and that was to remove Jetpack Related Posts from all posts within a category. This is what we came up with:
//Remove Jetpack Related Posts form Posts in a Category
if ( class_exists('Jetpack_RelatedPosts')) {

    function jetpackme_remove_rp() {
        $jprp = Jetpack_RelatedPosts::init();
        $callback = array($jprp, 'filter_add_target_to_dom');
        if (in_category(array(243))) {
            remove_filter('the_content', $callback, 40);
        }
    }

    add_action('wp', 'jetpackme_remove_rp', 20);
}
